src/pudl/transform/ferc1.py

```diff
@@ -1150,24 +1262,105 @@ def wide_to_tidy_xbrl(self, df: pd.DataFrame) -> pd.DataFrame:
         )
         return df
 
-    @cache_df("main")
-    def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
-        """Main transform.
+    def normalize_metadata_xbrl(
+        self, xbrl_fact_names: list[str] | None
+    ) -> pd.DataFrame:
+        """Transform the metadata to reflect the transformed data.
+
+        The XBRL Taxonomy metadata as extracted pertains to the XBRL data as extracted.
+        When we re-shape the data, we also need to adjust the metadata to be usable
+        alongside the reshaped data. For the plant in service table, this means
+        selecting metadata fields that pertain to the "stem" column name (not
+        differentiating between starting/ending balance, retirements, additions, etc.)
+
+        We fill in some gaps in the metadata, e.g. for FERC accounts that have been
+        split across multiple rows, or combined without being calculated. We also need
+        to rename the XBRL metadata categories to conform to the same naming convention
+        that we are using in the data itself (since FERC doesn't quite follow their own
+        naming conventions...). We use the same rename dictionary, but as an argument
+        to :meth:`pd.Series.replace` instead of :meth:`pd.DataFrame.rename`.
+        """
+        pis_meta = (
+            super()
+            .normalize_metadata_xbrl(xbrl_fact_names)
+            .assign(
+                ferc_account_label=lambda x: x.xbrl_fact_name.replace(
+                    self.params.rename_columns_instant_xbrl.columns
+                )
+            )
+        )
+
+        # Remove metadata records that pertain to columns we have eliminated through
+        # reshaping:
+        value_types = ["additions", "retirements", "adjustments", "transfers"]
+        pattern = ".*(" + "|".join(value_types) + ")$"
+        pis_meta = pis_meta[~pis_meta["ferc_account_label"].str.match(pattern)]
```

Comment:
your removing them because there is a stem-only column name

Reply:
Added a clarifying comment here.